File: VerveAPI/VerveAPI/Queue.m
//
//  Queue.m
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 6/1/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "Queue.h"

@implementation Queue

@synthesize queue = _queue;

- (id) init {
    self = [super init];
    if (self) {
        self.queue = [[NSMutableArray alloc] init];
    }
    return self;
}

- (void) push: (id) object {
    [self.queue addObject:object];
}
- (id) pop {
    id retval = [self.queue objectAtIndex:0];
    [self.queue removeObjectAtIndex:0];
    return retval;
}
@end

File: MDB_Swift/MyDailyBeat/API/src/Queue.swift
//  Converted with Swiftify v1.0.6242 - https://objectivec2swift.com/
//
//  Queue.swift
//  VerveAPI
//
//  Created by Virinchi Balabhadrapatruni on 6/1/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import Foundation
class Queue: NSObject {

    var queue = [Any]()

    override init() {
        super.init()
        
        self.queue = [Any]()
    
    }

    func push(_ object: Any) {
        self.queue.append(object)
    }

    func pop() -> Any {
        var retval: Any? = self.queue[0]
        self.queue.remove(at: 0)
        return retval!
    }
}